src/OpenTelemetry.AutoInstrumentation/Loading/LazyInstrumentationLoader.cs

```diff
@@ -47,34 +50,67 @@ private class OnAssemblyLoadInitializer
         private readonly ILifespanManager _lifespanManager;
         private readonly string _requiredAssemblyName;
 
+#if NETFRAMEWORK
+        private int _firstInitialization = 1;
```

Comment:
can you elaborate (preferably as comments in code) why these changes are needed?

Can  `AppDomain.CurrentDomain.AssemblyLoad` can be called from multiple threads in for the same assembly on .NET Framework?

Reply:
There are multiple important points: 
* In .NET Framework we don't have startup hook that executes before the main content. So the startup is injected to presumably to the beginning of the main method. 
* There is a race condition between checking if the assembly is already loaded and loading event. 
* There is a race condition after the determination logic, so checking logic is doubled with the event and pre-check logic

So basically, the check is doubled not to have any slight chance where `isRequiredAssemblyLoaded = false` and assembly load event is not subscribed.

Reply:
But for .NET it could also happen if one uses "another startup hook" which is executed before "OTel startup hook", right? I just think if he should not get rid of `#if NETFRAMEWORK`

Reply:
Also I suggest adding the important points as comments in code 😉

Reply:
Do you think that 1:1 is good enough or you would like to adjust the wording there?

Reply:
> But for .NET it could also happen if one uses "another startup hook" which is executed before "OTel startup hook", right? I just think if he should not get rid of #if NETFRAMEWORK

If you want to go there, I guess it gets deeper. I'm not sure if some would use different load context for eg, what happens then.

Reply:
> Do you think that 1:1 is good enough or you would like to adjust the wording there?

That is good enough for me 😉